src/stream/src/executor/mview/materialize.rs

```diff
@@ -776,62 +746,41 @@ impl<SD: ValueRowSerde> MaterializeCache<SD> {
     }
 
     fn evict(&mut self) {
-        self.data.evict()
+        self.lru_cache.evict()
     }
 }
 
-/// Generates a new row with the behavior of "do update if not null" by replacing columns in the old row with non-empty columns in the new row.
-///
-/// # Arguments
-///
-/// * `old_row` - The old row containing the original values.
-/// * `new_row` - The new row containing the replacement values.
+/// Replace columns in an existing row with the corresponding columns in a replacement row, if the
+/// column value in the replacement row is not null.
 ///
 /// # Example
 ///
-/// ```no_run
-/// let old_row = vec![Some(1), None, Some(3)];
-/// let mut new_row = vec![Some(10), Some(20), None];
-///
-///
-/// // After the execute_do_update_if_not_null_replacement function call, old_row will be [Some(10), Some(20), Some(3)].
+/// ```ignore
+/// let mut row = vec![Some(1), None, Some(3)];
+/// let replacement = vec![Some(10), Some(20), None];
+/// replace_if_not_null(&mut row, replacement);
 /// ```
-fn execute_do_update_if_not_null_replacement(
-    old_row: &mut Vec<Option<ScalarImpl>>,
-    new_row: OwnedRow,
-) {
-    for (old_col, new_col) in old_row.iter_mut().zip_eq_fast(new_row) {
+///
+/// After the call, `row` will be `[Some(10), Some(20), Some(3)]`.
+fn replace_if_not_null(row: &mut Vec<Option<ScalarImpl>>, replacement: OwnedRow) {
+    for (old_col, new_col) in row.iter_mut().zip_eq_fast(replacement) {
         if let Some(new_value) = new_col {
             *old_col = Some(new_value);
         }
     }
 }
-/// Determines whether pk conflict should be handled based on the version column of the new and old rows.
-///
-/// # Arguments
-///
-/// * `old_version_column`: The version column value of the old row.
-/// * `new_version_column`: The version column value of the new row.
-///
-/// # Returns
-///
-/// A boolean value indicating whether a conflict should be handled.
-///
-/// If both the new row's version column and the old row's version column are Some, the function will return true if the new row's version is greater than or equal to the old row's version.
-/// If the new row's version column is None and the old row's version column is Some, do not handle pk conflict.
-/// If both the new row's version column and the old row's version column are None, should handle pk conflict.
-fn should_handle_conflict(
-    old_version_column: &Option<ScalarImpl>,
-    new_version_column: &Option<ScalarImpl>,
+
+/// Determines whether pk conflict handling should update an existing row with newly-received value,
+/// according to the value of version column of the new and old rows.
```

Comment:
Can you also demonstrate what's `NULL` version is for?

Reply:
I think it's just in case that there's NULL value on the version field in some rows.

Reply:
ok maybe I should use `OrderType::nulls_smallest()`